dandiapi/api/management/commands/rename_ngff.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from django.db import transaction
+import djclick as click
+
+from dandiapi.api.models import ZarrArchive
+
+
+@click.command()
+@click.argument('dandiset_id', type=int, required=False)
+def rename_ngff(dandiset_id: int | None):
+    """
+    Change extension of all .ngff ZarrArchives (plus they're associated Asset paths) to .ome.zarr.
+
+    If DANDISET_ID is provided, only ZarrArchives in that dandiset will be modified.
+    Otherwise, every ZarrArchive in the DB is modified.
+    """
+    qs = ZarrArchive.objects.filter(name__endswith='.ngff')
+
+    # If a dandiset id was provided, filter by that.
+    if dandiset_id is not None:
+        click.echo(f'Renaming .ngff files to .ome.zarr for dandiset {dandiset_id}...')
+        qs = qs.filter(dandiset_id=dandiset_id)
+    else:
+        click.echo('Renaming .ngff files to .ome.zarr for all dandisets...')
+
+    for zarr in qs.prefetch_related('assets'):
+        new_name = f'{zarr.name[:-5]}.ome.zarr'
+
+        # only rename the ngff if a ome.zarr hasn't been uploaded already
+        if not ZarrArchive.objects.filter(name=new_name, dandiset=zarr.dandiset).exists():
+            click.echo(f'Renaming ZarrArchive ({zarr.id}) from {zarr.name} to {new_name}.')
+            with transaction.atomic():
+                zarr.name = new_name
+                zarr.save(update_fields=['name'])
+
+                for asset in zarr.assets.all():
```

Comment:
Should we use `.iterator` here?

Reply:
I suppose so, since a `ZarrArchive` could have a lot of associated assets, and the primary use case for this script would be running it in a `heroku run` environment (where we're constrained by the dyno's memory quota).